pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -366,12 +368,68 @@ public SegmentPartitionConfig getSegmentPartitionConfig() {
     }
   }
 
+  /**
+   * Get min time from the segment, based on the time column, only used by Kafka HLC.
+   */
+  @Deprecated
   public long getMinTime() {
-    return _minTime;
+    Long minTime = extractTimeValue(getMinVal(_timeColumnName));
+    if (minTime != null) {
+      return minTime;
+    }
+    return Long.MAX_VALUE;
   }
 
+  /**
+   * Get max time from the segment, based on the time column, only used by Kafka HLC.
+   */
+  @Deprecated
   public long getMaxTime() {
-    return _maxTime;
+    Long maxTime = extractTimeValue(getMaxVal(_timeColumnName));
+    if (maxTime != null) {
+      return maxTime;
+    }
+    return Long.MIN_VALUE;
+  }
+
+  private Long extractTimeValue(Comparable time) {
+    if (time != null) {
+      if (time instanceof Number) {
+        return ((Number) time).longValue();
+      } else {
+        String stringValue = time.toString();
+        if (StringUtils.isNumeric(stringValue)) {
```

Comment:
Is numeric is not good enough, recommend put try-catch over `Long.parseLong(stringValue)`

Reply:
I keep the same logic as before, this is for legacy HLC usage which assumes time is epoch value.

Reply:
nvm, I thought this is `NumberUtils.isNumber()`. This should be good